### ISSUE TITLE
BUG: Geod shared memory between objects

### ISSUE THREAD

**Issue by micahcochran** [jswhit/pyproj#64]
I was working on adding an \__eq__ operator (using the \__repr__ function in PR #63) for Geod, and found a shared memory bug.

For example (this is without the \__repr__ PR patch applied):
```Python
In [1]: from pyproj import Geod, pj_ellps

In [2]: pj_ellps['lerch']
Out[2]: {'a': 6378139.0, 'description': 'Lerch 1979', 'rf': 298.257}

In [3]: pj_ellps['clrk66']
Out[3]: {'a': 6378206.4, 'b': 6356583.8, 'description': 'Clarke 1866'}

# these ellipsoids are not equal
In [4]: pj_ellps['lerch']['a'] == pj_ellps['clrk66']['a']
Out[4]: False

In [5]: glerch = Geod(ellps='lerch')

In [6]: glerch.a
Out[6]: 6378139.0

In [7]: ga = glerch.a # store this

In [8]: repr(glerch)
Out[8]: '<pyproj.Geod object at 0x0208E090>'

In [9]: gclrk = Geod(ellps='clrk66')

In [10]: gclrk.a
Out[10]: 6378206.4

In [11]: repr(gclrk)
Out[11]: '<pyproj.Geod object at 0x0208E258>'

In [12]: ga == glerch.a  # compare those from #7, these should match
Out[12]: False

In [13]: glerch.a
Out[13]: 6378206.4  # this should be  6378139.0

In [14]: repr(glerch)
Out[14]: '<pyproj.Geod object at 0x0208E090>'

In [15]: repr(gclrk)
Out[15]: '<pyproj.Geod object at 0x0208E258>'

In [16]: glerch.a == gclrk.a  # should not match, see #4
Out[16]: True
```

The `.initstring` variable is unaffected by this.

There is some memory sharing happening in the Geod class between different instances.

One way to resolve this is just documenting that there can only be one Geod object.  I think it is preferable to rewriting this to allow for multiple Geod objects that do not share memory.

I tested pyproj version 1.9.5.1 and 1.9.3, both have the problem.  I'm planning to do a little more investigation.

**Comment by micahcochran** [comment]
This bug does not seem to have an impact the distance geodesic calculations.

IPython console:
```Python
In [1]: >>> from pyproj import Geod

In [2]: >>> g = Geod(ellps='clrk66') # Use Clarke 1966 ellipsoid.

In [3]: g.a
Out[3]: 6378206.4

In [4]: mercury = Geod(a=2439700) # Mercury

In [5]: mercury.a
Out[5]: 2439700

In [6]: g.a
Out[6]: 2439700

In [7]: >>> boston_lat = 42.+(15./60.); boston_lon = -71.-(7./60.)

In [8]: >>> portland_lat = 45.+(31./60.); portland_lon = -123.-(41./60.)

In [9]: >>> az12,az21,dist = g.inv(boston_lon,boston_lat,portland_lon,portland_l
at)

In [10]: dist
Out[10]: 4164192.7080994677

In [11]: >>> az12,az21,dist = mercury.inv(boston_lon,boston_lat,portland_lon,por
tland_lat)
In [12]: dist
Out[12]: 1590283.286640561

In [13]: >>> az12,az21,dist = g.inv(boston_lon,boston_lat,portland_lon,portland_
lat)

In [14]: dist
Out[14]: 4164192.7080994677
```

The distance calculations look smaller for Mercury, as expected.  I am very aware that Boston and Portland aren't on Mercury, but it illustrates the point.

I should have a PR soon that will fix this bug.

**Comment by micahcochran** [comment]
This was patched in PR #65.
